### PR TITLE
SEO changes in Documentation

### PR DESCRIPTION
**Pull request by kontsevoy**
* All documentation pages have dedicated <title> tag
* All documentation pages have dedicated <meta description> tag
* Fixed a few broken links
* Renamed some pages to make SEO-friendly URLs
* Found and updated all links to the renamed pages

### CODE REVIEW

**Review by benarent — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -232,7 +232,6 @@ docs-test-whitespace:
 .PHONY:docs-test-links
 docs-test-links: DOCS_FOLDERS := $(shell find . -name milv.config.yaml -exec dirname {} \;)
 docs-test-links:
-	go get -v github.com/magicmatatjahu/milv
```
Comment (webvictim) [inline Makefile:235]: This should be left in to install `milv` for people who don’t have it.

File: docs/4.2/milv.config.yaml [hunk 2]
```diff
@@ -29,6 +29,7 @@ white-list-external:
   - "https://teleport.example.com:8443"
   - "https://teleport.example.com"
   - "https://[route53_domain](#route53_domain"
+  - "https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-uhc-hmac-credentials-main"
```
Comment (webvictim) [inline docs/4.2/milv.config.yaml:32]: Why does this need whitelisting?

File: docs/4.3/quickstart.md [hunk 3]
```diff
@@ -11,22 +16,17 @@ VM.
...
 
-* We recommend that you read the [Architecture Guide](architecture/teleport_architecture_overview.md)
+* We recommend that you read the [Architecture Guide](architecture/overview.md)
   before working through this tutorial. If you'd like to dive right in though this is
   the best place to start!
 
 This guide is only meant to demonstrate how to run Teleport in a sandbox or demo
 environment, and showcase a few basic tasks you can do with Teleport.
 
-## Optional: Quickstart using Docker
```
Comment (webvictim) [inline docs/4.3/quickstart.md:21]: Where have the Docker instructions gone?

File: docs/4.2/milv.config.yaml (hunk 2 above)
Comment (benarent) [reply docs/4.2/milv.config.yaml:32]: It seems to be very 'flappy' / slow. We have done this perviously for docker links in 4.3 milv.

File: Makefile (hunk 1 above)
Comment (kontsevoy) [reply Makefile:235]: @webvictim The documentation building is Dockerized and milv is installed there. We're not installing mkdocs on every `make docs` do we?

Comment (webvictim) [reply Makefile:235]: Good point. Our `Makefile` targets aren't explicitly always inside a Docker container, but I don't know of a situation where we're running this separately at the moment.

**Review by webvictim — APPROVED** [review]

**Pull request merged**